### PR TITLE
Create test factories for user, events, and eras

### LINKED ISSUE #44: Create test factories for users, events, and eras
Would be nice to have an option to create a user with multiple events and eras automatically

### PR DESCRIPTION
**Pull request by leahriffell**
### Code Highlights:
 * Created test factories for users, events, and eras
 * Added ability to create a user with events and eras automatically
 * Added Faker gem
 * Replaced user variables in tests with factory users

### Where should the reviewer start?
 * spec/factories

### Have Tests Been Added?
- [x] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Issues:
* Closes #44 

### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [x] Rubocop Violations:

### CODE REVIEW

File: spec/factories/eras.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 FactoryBot.define do
   factory :era do
-    user { nil }
-    name { 'MyString' }
+    name { Faker::Hipster.sentence }
     start_date { '2020-12-19 15:37:22' }
```
Comment (leahriffell) [inline spec/factories/eras.rb:4]: Will wait to use Faker to generate dates until we know how we will store them (DateTime, int, string, etc.)

File: spec/requests/graphql/mutations/update_user_spec.rb [hunk 2]
```diff
@@ -5,9 +5,9 @@ module Users
     RSpec.describe UpdateUser, type: :request do
       describe '.resolve' do
         it 'updates a user' do
-          frosty = User.create!(name: 'Frosty', email: 'snow@brr.com', birthdate: DateTime.new(2001, 2, 3))
+          user = create(:user, :with_eras_events)
```
Comment (leahriffell) [inline spec/requests/graphql/mutations/update_user_spec.rb:8]: Syntax for creating a user with events and eras

File: spec/requests/graphql/queries/users/get_single_user_spec.rb [hunk 3]
```diff
@@ -3,9 +3,9 @@
 RSpec.describe Types::QueryType do
   describe 'display users' do
     it 'can query one user' do
-      user = User.create!(id: 1, name: 'Frosty', email: 'snow@brr.com', birthdate: DateTime.new(2001, 2, 3))
+      user = create(:user, :with_eras_events)
 
-      post graphql_path, params: { query: query }
+      post graphql_path, params: { query: query(id: user.id) }
```
Comment (leahriffell) [inline spec/requests/graphql/queries/users/get_single_user_spec.rb:8]: Make this query param dynamic

**Review by dhalverson — APPROVED** [review]
Looks good, excited to have hipster names

**Comment by dhalverson** [comment]
Factories look good, we can update once we decide on the DateTime piece!

**Pull request merged**